elastic/apm-server
# Instrument go-elasticsearch client

## Motivation/summary

This PR introduces instrumentation for the go-elasticsearch client, which is currently used for source maps and API Key auth. By capturing Elasticsearch queries, we can better diagnose performance issues involving Elasticsearch.

A couple of notes for reviewers:
 - Only API Key auth queries are fully instrumented at the moment. I will follow up later to propagate context through Transformables, so we can add spans for source mapping.
 - As this is not intended as a user-facing feature, I'm deferring system tests for now.

## How to test these changes

1. Enable the server's self-instrumentation, and API Key auth.
2. Instrument an application with an Elastic APM agent (I used the Go agent, naturally), configuring API Key auth
3. Observe in the APM app that "db.elasticsearch" spans are captured like so:

![image](https://user-images.githubusercontent.com/843579/74213441-2a057580-4cd4-11ea-9750-25b47253afd7.png)

## Related issues

Closes #3184
